.gitlab/dev_container_deploy/docker_linux.yml

```diff
@@ -225,12 +225,15 @@ single_machine_performance-nightly-amd64-a7:
   stage: dev_container_deploy
   rules:
     !reference [.on_deploy_nightly_repo_branch_a7]
+  only:
+    variables:
+      - $CI_COMMIT_BRANCH == "main"
```

Comment:
This is invalid, you can't mix `only:` and `rules:`
You'll need to add something like
```
- if $CI_COMMIT_BRANCH != main
  when: never
```
to the rules block